Add prometheus serviceMonitor manifest

dioguerra opened the pull request:
Adds the option to generate the prometheus serviceMonitor
manifest by setting .Values.metrics.serviceMonitor.enabled=true

Other configurations are also abvailable.

Signed-off-by: Diogo Guerra <diogo.filipe.tomas.guerra@cern.ch>

KeepMasterBranch commented:
Good luck, I wish this ever get merged, but so far, the owners of this project ignore pull requests from contributors

ninjadq commented:
Hi @dioguerra , Add serviceMonitor to harbor-helm is really useful. Could you also update the readme file to mention that install CRD of service monitor before enable it in value.yaml

dioguerra commented:
> Hi @dioguerra , Add serviceMonitor to harbor-helm is really useful. Could you also update the readme file to mention that install CRD of service monitor before enable it in value.yaml

All done. Not sure what you intended but i added 2 references onto the README and values.yaml

KeepMasterBranch commented on templates/metrics/metrics-svcmon.yaml:
```diff
@@ -0,0 +1,28 @@
+{{- if and ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) .Values.metrics.serviceMonitor.enabled }}
```
you should also have `.Values.metrics.enabled` too

dioguerra commented:
@ninjadq all good?

dirien commented:
hi,

can we get this approved? Is there anything i can help?

bclermont commented:
> can we get this approved?

no, harbor ignore their community of contributors, it had been like this for years

they rather internalize work and fund development instead of getting free contributions

vmware sucks at managing communities, same with other projects like velero

dirien commented:
@bclermont Sad, as its literally just the ServiceMonitor definition.  This is a CNCF graduated project and should reflect quicker the changes in the surrounding tool/infra landscape. 

I don't think, that using annotation based prometheus scraping os a thing anymore.

bclermont commented:
@dirien https://github.com/goharbor/community/issues/132

no one cares, you must fork this repo and maintains it to get fixes

ninjadq commented:
hi, @dioguerra sorry for the delayed reply. There are some conflicts here, need to resolve them first for the merge.

dioguerra commented:
> hi, @dioguerra sorry for the delayed reply. There are some conflicts here, need to resolve them first for the merge.

Rebased and fixed merge conflict on values.yaml

dioguerra commented:
@ninjadq

bclermont commented:
So, we just need to stir drama in all the other 50 pull requests to get a review?

jonasrosland commented:
@bclermont This was brought to my attention, and wanted to make sure you got a reply from us.

As community manager for Harbor, I am sad to see that you don't feel appreciated as a contributor. We do want contributors to feel welcome within the Harbor community, but that also applies to communications towards maintainers. 
We follow the CNCF CoC: https://github.com/cncf/foundation/blob/master/code-of-conduct.md and we want to make sure the comments here on GitHub, in Slack and other places follow the acceptable behavior.

However, your concerns are heard, contributions are always welcome, and we'll look over the issue and PR review times to see how this can be handled in a better manner.
